File: src/main/java/arc/arc/sync/base/Context.java
package arc.arc.sync.base;

import lombok.ToString;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

@ToString
public class Context {
    Map<String, Object> map = new HashMap<>();

    @SuppressWarnings("unchecked")
    public <T> T get(String key) {
        return (T) map.get(key);
    }

    public void put(String name, UUID uuid) {
        map.put(name, uuid);
    }
}

File: src/main/kotlin/ru/arc/sync/base/Context.kt
package ru.arc.sync.base

import java.util.UUID

class Context {
    private val map = HashMap<String, Any>()

    @Suppress("UNCHECKED_CAST")
    fun <T> get(key: String): T = map[key] as T

    fun put(name: String, uuid: UUID) {
        map[name] = uuid
    }
}
